### PR TITLE
Modexp memory context change

### PR DESCRIPTION
**Pull request by npwardberkeley**

### CODE REVIEW

**Review by wborgeaud — COMMENTED** [review]
Also need to update `evm/src/cpu/kernel/asm/core/precompiles/expmod.asm`, but otherwise looks good.

File: evm/src/cpu/kernel/asm/bignum/util.asm [hunk 1]
```diff
@@ -1,13 +1,15 @@
-%macro memcpy_kernel_general
+%macro memcpy_current_general
     // stack: dst, src, len
-    %stack (dst, src, len) -> (0, @SEGMENT_KERNEL_GENERAL, dst, 0, @SEGMENT_KERNEL_GENERAL, src, len, %%after)
+    GET_CONTEXT
+    %stack (context, dst, src, len) -> (context, @SEGMENT_KERNEL_GENERAL, dst, 0, @SEGMENT_KERNEL_GENERAL, src, len, %%after)
```
Comment (wborgeaud) [inline evm/src/cpu/kernel/asm/bignum/util.asm:4]: Not entirely sure about how these macros are used, but shouldn't the `0` be a `context`?

File: evm/src/cpu/kernel/asm/bignum/util.asm [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
     %jump(memcpy)
 %%after:
 %endmacro
 
-%macro clear_kernel_general
+%macro clear_current_general
     // stack: dst, len
-    %stack (dst, len) -> (0, @SEGMENT_KERNEL_GENERAL, dst, 0, len, %%after)
+    GET_CONTEXT
+    %stack (context, dst, len) -> (context, @SEGMENT_KERNEL_GENERAL, dst, 0, len, %%after)
```
Comment (wborgeaud) [inline evm/src/cpu/kernel/asm/bignum/util.asm:12]: ditto

File: evm/src/cpu/kernel/asm/bignum/util.asm (hunk 1 above)
Comment (npwardberkeley) [reply evm/src/cpu/kernel/asm/bignum/util.asm:4]: Ah good catch!

File: evm/src/cpu/kernel/asm/bignum/util.asm (hunk 2 above)
Comment (npwardberkeley) [reply evm/src/cpu/kernel/asm/bignum/util.asm:12]: This one is different, though: for tis `clear` macro, `0` is the value we're `memset`ing into the memory

File: evm/Cargo.toml [hunk 3]
```diff
@@ -26,8 +26,8 @@ num-bigint = "0.4.3"
 once_cell = "1.13.0"
 pest = "2.1.3"
 pest_derive = "2.1.0"
-plonky2 = { version = "0.1.2", default-features = false, features = ["timing"] }
-plonky2_util = { version = "0.1.0" }
+plonky2 = { path = "../plonky2", default-features = false, features = ["timing"] }
+plonky2_util = { path = "../util" }
```
Comment (wborgeaud) [inline evm/Cargo.toml:30]: Hm are these changes relevant to the PR?

Comment (npwardberkeley) [reply evm/Cargo.toml:30]: Oh sorry! That was so I could run the EVM tests, didn't mean to include it in the PR

**Review by wborgeaud — APPROVED** [review]
Looks good to me!
